hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieParquetWriter.java

```diff
@@ -84,7 +84,7 @@ public HoodieParquetWriter(String instantTime,
   }
 
   @Override
-  public void writeAvroWithMetadata(R avroRecord, HoodieRecord record) throws IOException {
+  public synchronized void writeAvroWithMetadata(R avroRecord, HoodieRecord record) throws IOException {
```

Comment:
This is problematic in a few ways:

1. This method is invoked in the hot-path, taking a lock in it would impact its performance considerably
2. Taking locks without timeouts exposes us to potential dead-locks

Reply:
Have removed the lock and use another way to guarantee parquet wrtier is used in a thread safe way